source/core/color.ts

```diff
@@ -0,0 +1,559 @@
+     * @returns - RGB color tuple: red, green, and blue, each in [0.0, 1.0].
+     */
+    static hsl2rgb(hsl: GLclampf3): GLclampf3 {
+        const hslF = this.clampf3(hsl, 'HSL input');
+
+        if (hslF[1] === 0.0) {
+            return [hslF[2], hslF[2], hslF[2]];
+        }
+
+        const q = hslF[2] < 0.5 ? hslF[2] * (1.0 + hslF[1]) : (hslF[2] + hslF[1]) - (hslF[1] * hslF[2]);
+        const p = 2.0 * hslF[2] - q;
+
+        return [Color.hue2rgb(p, q, hslF[0] + (1.0 / 3.0))
+            , Color.hue2rgb(p, q, hslF[0]), Color.hue2rgb(p, q, hslF[0] - (1.0 / 3.0))];
+    }
+
+    /**
+     * Converts a color from HSL space to RGB space.
+     * @param rgb - RGB color tuple: red, green, and blue, each in [0.0, 1.0].
+     * @returns - HSL color tuple: hue, saturation, and lightness, each in [0.0, 1.0].
+     */
+    static rgb2hsl(rgb: GLclampf3): GLclampf3 {
+        const rgbF = this.clampf3(rgb, 'RGB input');
+        const hsl: GLclampf3 = [0.0, 0.0, 0.0];
+
+        const min = Math.min(rgbF[0], rgbF[1], rgbF[2]);
+        const max = Math.max(rgbF[0], rgbF[1], rgbF[2]);
+        const delta = max - min;
+
+        hsl[2] = (max + min) * 0.5;
+
+        if (delta === 0.0) {
+            return hsl;
+        }
+
+        hsl[1] = hsl[2] < 0.5 ? delta / (max + min) : delta / (2.0 - max - min);
+
+        const deltaR = (((max - rgbF[0]) / 6.0) + (delta / 2.0)) / delta;
+        const deltaG = (((max - rgbF[1]) / 6.0) + (delta / 2.0)) / delta;
+        const deltaB = (((max - rgbF[2]) / 6.0) + (delta / 2.0)) / delta;
+
+        if (rgbF[0] === max) {
+            hsl[0] = deltaB - deltaG;
+        } else if (rgbF[1] === max) {
+            hsl[0] = deltaR - deltaB + (1.0 / 3.0);
+        } else { // if (rgbF[2] === max) {
+            hsl[0] = deltaG - deltaR + (2.0 / 3.0);
+        }
+        return hsl;
+    }
+
+
+    /**
+     * Converts a color from LAB space to XYZ space.
+     * @param lab - LAB color tuple: lightness, greenRed, and blueYellow, each in [0.0, 1.0].
+     * @returns - XYZ color tuple: x, y, and z, each in [0.0, 1.0].
+     */
+    static lab2xyz(lab: GLclampf3): GLclampf3 {
+        const labF = this.clampf3(lab, 'LAB input');
+
+        const yr = (labF[0] * 100.0 + 16.0) / 116.0;
+        const xr = labF[1] * 100 / 500.0 + yr;
+        const zr = yr - labF[2] * 100 / 200.0;
+
+        const yr3 = yr * yr * yr;
+        const xr3 = xr * xr * xr;
+        const zr3 = zr * zr * zr;
+
+        /* D65/2° illuminant for XYZ conversion */
+        const y = 1.00000 * (yr3 > 0.008856 ? yr3 : (yr - 16.0 / 116.0) / 7.787);
+        const x = 0.95047 * (xr3 > 0.008856 ? xr3 : (xr - 16.0 / 116.0) / 7.787);
+        const z = 1.08883 * (zr3 > 0.008856 ? zr3 : (zr - 16.0 / 116.0) / 7.787);
+
+        return [x, y, z];
+    }
+
+    /**
+     * Converts a color from XYZ space to LAB space.
+     * @param xyz- XYZ color tuple: x, y, and z, each in [0.0, 1.0].
```

Comment:
Minor: missing space between `xyz` and `-` (comment doesn't show up in the generated docs).